### PR TITLE
correct the 1870 connection order for multiple corporations

### LINKED ISSUE #7145: [1870] Simultaneous Connection Runs not always in correct order
Split out from #5847
https://18xx.games/game/37799?action=598
The connection runs are occurring in the wrong order. When multiple companies become eligible at the same time, the currently operating company goes first, followed by the other companies in stock price order (p. 22). Which in this case would be MP, ATSF, MKT.

### PR DESCRIPTION
**Pull request by JoeB989**
When multiple companies become eligible at the same time, the currently operating company goes first, followed by the other companies in stock price order (p. 22).
Fixes #7145

### CODE REVIEW

**Comment by tobymao** [comment]
why is this a  draft?

**Comment by JoeB989** [comment]
I screwed up my branch base so I need previous PR complete and merged back into my branch before starting this PR

File: lib/engine/game/g_1870/step/check_connection.rb [hunk 1]
```diff
@@ -15,18 +15,22 @@ class CheckConnection < Engine::Step::Base
...
           def auto_actions(entity)
             corporations = if @round.entity_index.zero? || @round.step_passed?(G1870::Step::CheckConnection)
-                             @game.corporations.select { |c| destination(c) }
+                             @round.entities.select { |c| destination?(c) } # destinate in OR order
                            else
                              []
                            end
 
+            # if the current corporation is also destinating, it must run first
+            i = corporations.find_index(@round.current_operator)
```
Comment (tobymao) [inline lib/engine/game/g_1870/step/check_connection.rb:24]: just index

File: lib/engine/game/g_1870/step/check_connection.rb [hunk 2]
```diff
@@ -15,18 +15,22 @@ class CheckConnection < Engine::Step::Base
...
+            corporations = [@round.current_operator] + (corporations - [@round.current_operator]) if !i.nil? && i.positive?
```
Comment (tobymao) [inline lib/engine/game/g_1870/step/check_connection.rb:25]: if i&.positive?

**Review by tobymao — APPROVED** [review]

**Pull request merged**